flang/lib/Frontend/CompilerInvocation.cpp

```diff
@@ -248,13 +248,21 @@ static void parseCodeGenArgs(Fortran::frontend::CodeGenOptions &opts,
   if (const llvm::opt::Arg *a =
           args.getLastArg(clang::driver::options::OPT_mframe_pointer_EQ)) {
     llvm::StringRef s = a->getValue();
-    assert(s == "none" || s == "non-leaf" || s == "all");
-    if (s == "none")
+    
+    if (!(s=="none" || s=="no-leaf" || s=="all")) {
+    const auto debugWarningId = diags.getCustomDiagID(
+        clang::DiagnosticsEngine::Warning, "Frame pointer: %0");
+    diags.Report(debugWarningId).AddString(a->getValue());
+    }
+
+    if (s == "none") {
       opts.setFramePointer(llvm::FramePointerKind::None);
-    else if (s == "non-leaf")
-      opts.setFramePointer(llvm::FramePointerKind::NonLeaf);
-    else
-      opts.setFramePointer(llvm::FramePointerKind::All);
+    } else {
+      if (s == "non-leaf")
+        opts.setFramePointer(llvm::FramePointerKind::NonLeaf);
+      else
+        opts.setFramePointer(llvm::FramePointerKind::All);
```

Comment:
This is a change suggested above by Kiran. 
https://github.com/llvm/llvm-project/pull/74598#discussion_r1417479590

Reply:
Ahh I missread the old side of the diff. What I liked would be something more like
```
if (s == "option1")
  ...
else if (s == "option2")
  ...
else if (s == "option3")
  ...
else
  error!
```

There is no need for braces on any of these because there will only be a single line in the body of the if statements.

Kiran was right that in this case
```
if (cond)
  something
else
  if (cond2)
     foo
  else
    bar
```

It is a good idea to put braces around the outer if because otherwise it is harder to see to which `if` the final `else` belongs.

Reply:
I agree with @tblah that they should all be at the same indentation. The old formattting (copied below) made it look like there are two levels of indentation and hence my comment. 
```
    if (s == "none")
      opts.setFramePointer(llvm::FramePointerKind::None);
    else
      if (s == "non-leaf")
        opts.setFramePointer(llvm::FramePointerKind::NonLeaf);
      else
        opts.setFramePointer(llvm::FramePointerKind::All);
 ```

Reply:
Ok, I will then update this in the following commit.